### PR TITLE
cherry picks for 0.17.4

### PR DESCRIPTION
**Pull request by johnynek**
Bring all the binary compatible changes from develop into 0.17 branch

binary compatibility checked:
```
[info] scalding-core: found 0 potential binary incompatibilities while checking against com.twitter:scalding-core_2.11:0.17.2
[info] scalding-avro: found 0 potential binary incompatibilities while checking against com.twitter:scalding-avro_2.11:0.17.2
[info] scalding-hadoop-test: found 0 potential binary incompatibilities while checking against com.twitter:scalding-hadoop-test_2.11:0.17.2
[info] scalding-repl: found 0 potential binary incompatibilities while checking against com.twitter:scalding-repl_2.11:0.17.2
[info] scalding-json: found 0 potential binary incompatibilities while checking against com.twitter:scalding-json_2.11:0.17.2
[info] scalding-db: found 0 potential binary incompatibilities while checking against com.twitter:scalding-db_2.11:0.17.2
[info] scalding-hraven: found 0 potential binary incompatibilities while checking against com.twitter:scalding-hraven_2.11:0.17.2
[info] execution-tutorial: previous-artifact not set, not analyzing binary compatibility
[info] scalding-jdbc: found 0 potential binary incompatibilities while checking against com.twitter:scalding-jdbc_2.11:0.17.2
[info] scalding-parquet: found 0 potential binary incompatibilities while checking against com.twitter:scalding-parquet_2.11:0.17.2
[info] scalding-commons: found 0 potential binary incompatibilities while checking against com.twitter:scalding-commons_2.11:0.17.2
[info] scalding-thrift-macros: found 0 potential binary incompatibilities while checking against com.twitter:scalding-thrift-macros_2.11:0.17.2
[info] scalding-parquet-scrooge: found 0 potential binary incompatibilities while checking against com.twitter:scalding-parquet-scrooge_2.11:0.17.2
[success] Total time: 12 s, completed Dec 19, 2017 9:46:39 AM
```

cc @FlavSF @moulimukherjee @snoble @pankajroark

Will publish 0.17.4 after this is merged.

### CODE REVIEW

File: scalding-core/src/main/scala/com/twitter/scalding/FileSource.scala [hunk 1]
```diff
@@ -229,7 +229,13 @@ abstract class FileSource extends SchemedSource with LocalSourceOverride with Hf
    * TODO: consider writing a more in-depth version of this method in [[TimePathedSource]] that looks for
    * TODO: missing days / hours etc.
    */
-  protected def pathIsGood(globPattern: String, conf: Configuration) = FileSource.globHasNonHiddenPaths(globPattern, conf)
+  protected def pathIsGood(globPattern: String, conf: Configuration) = {
+    if (conf.getBoolean("scalding.require_success_file", false)) {
```
Comment (ianoc) [inline scalding-core/src/main/scala/com/twitter/scalding/FileSource.scala:233]: uh this seems fine given this is a cherrypick pr... but probably should be using a constant?

**Comment by ianoc** [comment]
lgtm

File: scalding-core/src/main/scala/com/twitter/scalding/FileSource.scala (hunk 1 above)
Comment (johnynek) [reply scalding-core/src/main/scala/com/twitter/scalding/FileSource.scala:233]: we didn't want to break binary compatibility, but a constant would be a nice follow up.

**Comment by johnynek** [comment]
releasing 0.17.4

**Pull request merged**